### PR TITLE
12 school evaluation Rating model improvements

### PR DESCRIPTION
**Pull request by AnuKPatel**
Removed the rating.rating model defined in school_evaluation module and used the base rating module's rating.rating by inherit it.

Kindly review and merge the code.

### CODE REVIEW

File: school_evaluation/models/school_evaluation.py [hunk 1]
```diff
@@ -164,21 +163,33 @@ class SchoolEvaluationTemplate(models.Model):
...
+    @api.depends('res_model', 'res_id')
+    def _compute_res_name(self):
+        # cannot change the rec_name of session since it is use to create the bus channel
+        # so, need to override this method to set the same alternative rec_name as rating
+        if self.res_model == 'base':
+            self.res_name = self.rating
+        else:
+            super(RatingRating, self)._compute_res_name()
 
     rating_id = fields.Many2one('school.evaluation.template', 'Stud',
```
Comment (JayVora-SerpentCS) [inline school_evaluation/models/school_evaluation.py]: field name sounds illogical. plz make it meaningful.

File: school_evaluation/models/school_evaluation.py [hunk 2]
```diff
@@ -164,21 +163,33 @@ class SchoolEvaluationTemplate(models.Model):
...
-    _description = "Rating details"
-    _rec_name = 'point'
-    _order = "point desc"
+    _inherit = 'rating.rating'
+
+    @api.model
+    def create(self, vals):
+        res_model = self.env['school.evaluation.template'].browse(
+            vals.get('res_id'))
+        vals.update({'res_model_id': res_model.id})
```
Comment (JayVora-SerpentCS) [inline school_evaluation/models/school_evaluation.py:177]: mention the reason plz

Comment (Freni-SerpentCS) [reply school_evaluation/models/school_evaluation.py:177]: to set Document Model (res_model_id)  which is mandatory for the rating(which is inherited from rating.rating).

Comment (JayVora-SerpentCS) [reply school_evaluation/models/school_evaluation.py:177]: Did you mean res_id is different than res_model_id?
I see in the code k you fetch MODEL from  ID and set ID.. Whats the difference?
@AnuKPatel

Comment (AnuKPatel) [reply school_evaluation/models/school_evaluation.py:177]: @Freni-SerpentCS  res_model_id is as 'database id' of the model, Like school_evaluation_template is stored in ir_model table and it's record have id.

res_id is an id of the school_evaluation_template. For Ex. Rating is belongs to 4rth no. evaluation template.

so you need to improve the code here.

Comment (JayVora-SerpentCS) [reply school_evaluation/models/school_evaluation.py:177]: Was there a code done afterwards?

File: school_evaluation/models/school_evaluation.py [hunk 3]
```diff
@@ -164,20 +163,33 @@ class SchoolEvaluationTemplate(models.Model):
...
+    @api.model
+    def create(self, vals):
+        """Set Document model name for rating."""
+        res_model = self.env['ir.model'].search([
+            ('model', '=', 'school.evaluation.template')])
+        vals.update({'res_model_id': res_model.id})
+        res = super(RatingRating, self).create(vals)
+        return res
+
+    @api.one
```
Comment (JayVora-SerpentCS) [inline school_evaluation/models/school_evaluation.py:181]: sure api.one?

File: school_evaluation/models/school_evaluation.py [hunk 4]
```diff
@@ -164,20 +163,33 @@ class SchoolEvaluationTemplate(models.Model):
...
+    @api.depends('res_model', 'res_id')
+    def _compute_res_name(self):
+        # cannot change the rec_name of session since it is use to create the bus channel
+        # so, need to override this method to set the same alternative rec_name as rating
+        if self.res_model == 'school.evaluation.template':
+            self.res_name = self.rating
+        else:
+            super(RatingRating, self)._compute_res_name()
+
+    template_id = fields.Many2one('school.evaluation.template', 'Stud',
```
Comment (JayVora-SerpentCS) [inline school_evaluation/models/school_evaluation.py]: Stud isnt a nice string for the Template

File: school_evaluation/models/school_evaluation.py (hunk 1 above)
Comment (AnuKPatel) [reply school_evaluation/models/school_evaluation.py]: Improved now.

File: school_evaluation/models/school_evaluation.py (hunk 2 above)
Comment (AnuKPatel) [reply school_evaluation/models/school_evaluation.py:177]: yes, Improvements are done.

File: school_evaluation/models/school_evaluation.py (hunk 3 above)
Comment (Freni-SerpentCS) [reply school_evaluation/models/school_evaluation.py:181]: Yes. This method is overridden from the base and there's api.one.

**Pull request merged**